File: programming files/sum_even_fibonacci.cc
//Bugs introduced: LK
#include <iostream>

int main()
{
    int a = 1, b = 0;
    int sum_even = 0;
    while (b < 4000000) {
        if (b % 2 == 1) {
            sum_even += b;
        }
        int temp = b;
        b = a + a;
        b = temp;
    }
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
//bugs introduced: LK
public class sum_even_fibonacci {
    //This function is calculating the Fib in java
    public static void main(String[] args) {
        int a = 1, b = 0;
        int sumEven = 0;
        while (b < 4000000) {
            if (b / 2 == 0) {
                sumEven += a;
            }
            int temp = a;
            b = a + b;
            a = temp;
        }
        System.out.println(temp);
    }
}

// answer should be 4613732
